Update package documentation (#7)
ballerina-platform/module-ballerinax-ibm.ctg: README.md

[![GitHub Last Commit](https://img.shields.io/github/last-commit/ballerina-platform/module-ballerinax-ibm.ctg.svg)](https://github.com/ballerina-platform/module-ballerinax-ibm.ctg/commits/main)
[![GitHub Issues](https://img.shields.io/github/issues/ballerina-platform/ballerina-library/module/ibm.ctg.svg?label=Open%20Issues)](https://github.com/ballerina-platform/ballerina-library/labels/module%2Fibm.ctg)

[IBM CICS Transaction Gateway (CTG)](https://www.ibm.com/products/cics-transaction-gateway) is a robust middleware solution that facilitates communication between distributed applications and IBM CICS Transaction Servers.

The `ballerinax/ibm.ctg` package provides an API to connect to an IBM CICS Transaction Gateway using Ballerina. The current connector is compatible with IBM CICS Transaction Gateway versions 9.3.

## Quickstart

To use the IBM CTG connector in your Ballerina application, modify the `.bal` file as follows:

### Step 1: Import the connector

Import `ballerinax/ibm.ctg` module into your Ballerina project.

```ballerina
import ballerinax/ibm.ctg;
```

### Step 2: Add IBM MQ driver

> Add `ccf2`, `cicsjee.`, and `ctgclient` as a platform dependency to the `Ballerina.toml`.

Reviewer: ```suggestion
Add `ccf2`, `cicsjee.`, and `ctgclient` as platform dependencies to the `Ballerina.toml`.
```

Author: Fixed with [b90726216354161233a837b8adaeb27190a77f91](https://github.com/ballerina-platform/module-ballerinax-ibm.ctg/pull/7/commits/b90726216354161233a837b8adaeb27190a77f91)

```toml
[[platform.java17.dependency]]
groupId = "ibm"
artifactId = "ccf2"
version = "9.3"
path = "./<path-to>/ccf2.jar"

[[platform.java17.dependency]]
groupId = "ibm"
artifactId = "cicsjee"
version = "9.3"
path = "./<path-to>/cicsjee.jar"

[[platform.java17.dependency]]
groupId = "ibm"
artifactId = "ctgclient"
version = "9.3"
path = "./<path-to>/ctgclient.jar"
```

### Step 3: Instantiate a new connector

Create an `ctg:Client` instance by giving IBM MQ configuration.

```ballerina
configurable string host = ?;
configurable int port = ?;
configurable string cicsServer = ?;
configurable string userId = ?;
configurable string password = ?;

ctg:Client ctg = check new({
    host,
    port,
    cicsServer,
    auth: {
        userId,
        password
    }
});
```

### Step 4: Invoke the connector operations

Now, utilize the available connector operations.

```ballerina
string programName = ...;

> byte[] inputPayload = ...;

Reviewer: IMO better to add comments to the incomplete statements, as user might try to copy paste the content and run as it is otherwise

Author: Fixed with [b90726216354161233a837b8adaeb27190a77f91](https://github.com/ballerina-platform/module-ballerinax-ibm.ctg/pull/7/commits/b90726216354161233a837b8adaeb27190a77f91)

[![GitHub Last Commit](https://img.shields.io/github/last-commit/ballerina-platform/module-ballerinax-ibm.ctg.svg)](https://github.com/ballerina-platform/module-ballerinax-ibm.ctg/commits/main)
[![GitHub Issues](https://img.shields.io/github/issues/ballerina-platform/ballerina-library/module/ibm.ctg.svg?label=Open%20Issues)](https://github.com/ballerina-platform/ballerina-library/labels/module%2Fibm.ctg)

[IBM CICS Transaction Gateway (CTG)](https://www.ibm.com/products/cics-transaction-gateway) is a robust middleware solution that facilitates communication between distributed applications and IBM CICS Transaction Servers.

> The `ballerinax/ibm.ctg` package provides an API to connect to an IBM CICS Transaction Gateway using Ballerina. The current connector is compatible with IBM CICS Transaction Gateway versions 9.3.

Reviewer 2: ```suggestion
The `ballerinax/ibm.ctg` package provides an API to connect to an IBM CICS Transaction Gateway using Ballerina. The current connector is compatible with IBM CICS Transaction Gateway version 9.3.
```

## Quickstart

To use the IBM CTG connector in your Ballerina application, modify the `.bal` file as follows:

### Step 1: Import the connector

Import `ballerinax/ibm.ctg` module into your Ballerina project.

```ballerina
import ballerinax/ibm.ctg;
```

### Step 2: Add IBM MQ driver

> Add `ccf2`, `cicsjee.`, and `ctgclient` as a platform dependencies to the `Ballerina.toml`.

Reviewer 2: ```suggestion
Add `ccf2`, `cicsjee`, and `ctgclient` as a platform dependencies to the `Ballerina.toml`.
```

Author: Fixed with [e1463f1a6e02df6874fb5f8bf141068ee3028271](https://github.com/ballerina-platform/module-ballerinax-ibm.ctg/pull/7/commits/e1463f1a6e02df6874fb5f8bf141068ee3028271)

```toml
[[platform.java17.dependency]]
groupId = "ibm"
artifactId = "ccf2"
version = "9.3"
path = "./<path-to>/ccf2.jar"

[[platform.java17.dependency]]
groupId = "ibm"
artifactId = "cicsjee"
version = "9.3"
path = "./<path-to>/cicsjee.jar"

[[platform.java17.dependency]]
groupId = "ibm"
artifactId = "ctgclient"
version = "9.3"
path = "./<path-to>/ctgclient.jar"
```

### Step 3: Instantiate a new connector

Create an `ctg:Client` instance by giving IBM MQ configuration.

```ballerina
configurable string host = ?;
configurable int port = ?;
configurable string cicsServer = ?;
configurable string userId = ?;
configurable string password = ?;

ctg:Client ctg = check new({
    host,
    port,
    cicsServer,
    auth: {
        userId,
        password
    }
});
```

### Step 4: Invoke the connector operations

Now, utilize the available connector operations.

```ballerina
// set the CICS program name here
string programName = ...;
// set the input payload
byte[] inputPayload = ...;
byte[]? outputPayload = ctg->execute(programName = programName, commArea = inputPayload);
```

### Step 5: Run the Ballerina application

> ```Shell
> bal run
> ```

Reviewer 2: ```suggestion
```shell
bal run
```
```

Author: Fixed with [e1463f1a6e02df6874fb5f8bf141068ee3028271](https://github.com/ballerina-platform/module-ballerinax-ibm.ctg/pull/7/commits/e1463f1a6e02df6874fb5f8bf141068ee3028271)
